### ISSUE TITLE
Support Gaussian splatting for Turntable captures

### ISSUE THREAD

**Issue by jb-ye** [nerfstudio-project/nerfstudio#3327]
**Is your feature request related to a problem? Please describe.**

Normal Nerf/GS methods usually don't support turntable captures out of box. However, it is a very common setup for traditional photogrammetry datasets. 

**Describe the solution you'd like**

TBD

If you are interested in this feature, please vote in this issue and possibly share your datasets from turntable captures to assist the development.

**Comment by pgavriel** [comment]
I have 5 cameras set up around a turntable, and the best solution I've found so far is to precompute all the "virtual" camera positions and generate the transforms.json myself with a script that mimics the format generated by ns-process-data.  

To do this I first use COLMAP to determine the relative transforms between all the cameras for a single turntable position, and I use a feature dense calibration image on the turntable to do this.  
I then load COLMAPs dense reconstruction into cloud compare to level the turntable and translate it so that the turntable center is at (0,0,0) and use that transform as my "alignment transform".  
Once all these aligned camera transforms are calculated, you can perform a z-axis rotational transform on each camera to get it's virtual position for any turntable position.  
I've been able to get pretty good NeRF and GS results from turntable images using this method. I'll add the calibration image I use. 
![MOAD_calib_small](https://github.com/user-attachments/assets/2af7e6a5-918a-4df9-977a-b4ee41df46ae)

**Comment by jb-ye** [comment]
@pgavriel Thanks for following up on this thread. Something isn't clear to me though:

Once you've determined the relative poses between your five cameras, how do you calculate their relative poses of camera to each individual turntable position during the real object capture? Are you using a rigid rig assumption? Does COLMAP support multi-camera with known extrinsics?

**Comment by pgavriel** [comment]
No problem! 

The rig I'm using is rigid between the turntable and the camera arm so that is assumed. 
![20240801_163721](https://github.com/user-attachments/assets/f7628020-9efc-47fe-97ef-3ab548685fe7)

The turntable is controlled precisely by a stepper motor to known degree positions before data capture, so by knowing the positions I intend to capture, I can calculate the transforms for those positions like so: 
![moad_colmap_scan-example](https://github.com/user-attachments/assets/e852b70c-fd5e-4fdf-8718-d8b9946cb274)
As long as the alignment transform correctly levels and centers the turntable on the origin, you just need to rotate about the z-axis to find the camera transforms for a given turntable position. 

And I do believe COLMAP has the capability to handle different camera types simultaneously if that is your question

**Comment by jb-ye** [comment]
@pgavriel thanks that being said in your setup, the motor control has to be very accurate in terms of degrees? Do you know its specs in general?

This setting has high requirement on the underlying hardware and not all people can build this easily. That being said, one would want to support more generic setting, for example, capture an object on a microwave oven plate using a set of fixed cameras.

**Comment by pgavriel** [comment]
@jb-ye Yes for my application I'm using a pretty expensive stepper motor setup that allows for very precise turntable movement, I believe the model is a PX110-200. I still believe you could get decent results with a less precise motor, maybe if you had markings on the turntable and you could determine the position based on image data or use some kind of encoder, this kind of setup could still work on a smaller budget. Best of luck!

**Comment by SharkWipf** [comment]
(Any motor is precise if you use a large enough gear ratio)
